cordova-lib/src/platforms/PlatformApi.js

```diff
@@ -0,0 +1,414 @@
+ *                               plugin_id: plugin id, such as 'org.apache.cordova.camera'
+ *                               options: complex object, passed by plugman.
+ *                                   See plugman/install module for reference.
+ *                               project: object that represents underlying IDE project for plaform.
+ *                                   For Android it's an instance of plugman/util/android-project,
+ *                                   for Windows - util/windows/jsprojmanager, etc.
+ */
+BasePlatformApi.prototype.getInstaller = function(type) {
+    var self = this;
+    function installWrapper(item, plugin_dir, plugin_id, options, project) {
+        self.getPluginHandler()[type].install(item, plugin_dir, self.root, plugin_id, options, project);
+    }
+    return installWrapper;
+};
+
+/**
+ * Gets an uninstaller function, that handles uninstallation for specific type of plugin's files
+ * (such as <src-file>, <lib-file>, etc.). Used in plugman only.
+ * @param  {String} type Plugin file type (<src-file>, <lib-file>, etc.)
+ * @return {Function}    Handler function for file, that accepts the following set of arguments:
+ *                               item: corresponding item from plugin.xml file.
+ *                                   Could be get by PluginInfo.get* methods.
+ *                               plugin_id: plugin id, such as 'org.apache.cordova.camera'
+ *                               options: complex object, passed by plugman.
+ *                                   See plugman/install module for reference.
+ *                               project: object that represents underlying IDE project for plaform.
+ *                                   For Android it's an instance of plugman/util/android-project,
+ *                                   for Windows - util/windows/jsprojmanager, etc.
+ */
+BasePlatformApi.prototype.getUninstaller = function(type) {
+    var self = this;
+    function uninstallWrapper(item, plugin_id, options, project) {
+        self.getPluginHandler()[type].uninstall(item, self.root, plugin_id, options, project);
+    }
+    return uninstallWrapper;
+};
+
+/**
+ * Base implementation for getConfigXml. Assumes that config.xml
+ * is placed at the root of project.
+ * @return {String} Path to platform's config.xml file
+ */
+BasePlatformApi.prototype.getConfigXml = function() {
+    if (this.parser && this.parser.config_xml) {
+        return this.parser.config_xml();
+    }
+
+    return  path.join(this.root, 'config.xml');
+};
+
+/**
+ * Base implementation for updateConfig. Reset platform's config to default.
+ * @return {String} Path to platform's config.xml file
+ */
+BasePlatformApi.prototype.updateConfig = function(optSource) {
+    var defaultConfig = path.join(this.root, 'cordova', 'defaults.xml');
+    var ownConfig = this.getConfigXml();
+    // If defaults.xml is present, overwrite platform config.xml with it
+    if (fs.existsSync(defaultConfig)) {
```

Comment:
The old version of the code also handled the case when both defaults.xml and project_dir/config.xml were not present.
In that case, it would grab 'www/config.xml' and copy it to project_dir/config.xml.

Any reason that case isn't handled anymore ?